src/cplscheme/CouplingScheme.cpp

```diff
@@ -8,7 +8,7 @@ const int CouplingScheme::UNDEFINED_TIME_WINDOWS = -1;
 
 const double CouplingScheme::UNDEFINED_TIME_WINDOW_SIZE = -1.0;
 
-const int CouplingScheme::UNDEFINED_EXTRAPOLATION_ORDER = -1;
+const int CouplingScheme::UNDEFINED_EXTRAPOLATION_ORDER = 0; // by default use zeroth order extrapolation.
```

Comment:
Why this change? Shouldn't the configuration handle this?
At least, the name of the variable is confusing now.

Reply:
I changed this a bit in cb867ca3b11c66b8ece01def1abbfce11296f387. Does this look better?

Why do we need this? We need to always perform an extrapolation now, because we always extrapolate, if we move (in most cases just constant). Previously we did not always perform the extrapolation. But I agree: It's reasonable to be able to differentiate between no extrapolation and zeroth order extrapolation (explicitly set in the config).

Reply:
Looks better already. A bit nicer could be if the order is always set in the configuration. If the user doesn't set it, it defaults to zero. I thought this is how we handled it so far as well. Also before waveform things, we always extrapolated at least at order zero.